aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/UpdateHandler.java

```diff
@@ -45,31 +46,93 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final Logger logger) {
         this.logger = logger;
         final ResourceModel model = request.getDesiredResourceState();
+        TagHelper tagHelper = new TagHelper();
 
         if(model.getType().equalsIgnoreCase(ParameterType.SECURE_STRING.toString())) {
             String message = String.format("SSM Parameters of type %s cannot be updated using CloudFormation", ParameterType.SECURE_STRING);
             return ProgressEvent.defaultFailureHandler(new TerminalException(message),
                     HandlerErrorCode.InvalidRequest);
         }
 
-        return ProgressEvent.progress(model, callbackContext)
-                // First validate the resource actually exists per the contract requirements
-                // https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
-                .then(progress ->
-                        proxy.initiate("aws-ssm-parameter::validate-resource-exists", proxyClient, model, callbackContext)
-                                .translateToServiceRequest(Translator::getParametersRequest)
-                                .makeServiceCall(this::validateResourceExists)
-                                .progress())
+        ProgressEvent<ResourceModel, CallbackContext> progressEvent = ProgressEvent.progress(model, callbackContext);
 
-                .then(progress ->
-                        proxy.initiate("aws-ssm-parameter::resource-update", proxyClient, model, callbackContext)
-                                .translateToServiceRequest(Translator::updatePutParameterRequest)
-                                .backoffDelay(getBackOffDelay(model))
-                                .makeServiceCall(this::updateResource)
-                                .stabilize(BaseHandlerStd::stabilize)
-                                .progress())
-                .then(progress -> handleTagging(proxy, proxyClient, progress, model, request.getDesiredResourceTags(), request.getPreviousResourceTags()))
-                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+        if (TagHelper.shouldUpdateTags(request)) {
```

Comment:
Looking at the previous commit, you'd:

1. validate if parameter exists
2. update the parameter if necessary.

With this change, if there are tags change, you won't try to validate if the SSM parameter exists first. Is this expected?